apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -441,25 +450,147 @@ export const getResourceFullPermalink = async (
   return `/${permalinkTree.join("/")}`
 }
 
-export const publishResource = async (
+export const publishPageResource = async (
   logger: Logger<string>,
   siteId: number,
   resourceId: string,
   userId: string,
 ) => {
   // Step 1: Create a new version
-  const addedVersionResult = await incrementVersion({
-    siteId,
-    resourceId,
-    userId,
-  })
+  const by = await db
+    .selectFrom("User")
+    .selectAll()
+    .where("id", "=", userId)
+    .executeTakeFirstOrThrow(
+      () =>
+        new TRPCError({
+          code: "PRECONDITION_FAILED",
+          message: "Please ensure that you have logged in",
+        }),
+    )
+
+  const addedVersionResult = await db
+    .transaction()
+    .setIsolationLevel("serializable")
+    .execute(async (tx) => {
+      const fullResource = await getFullPageById(tx, {
+        resourceId: Number(resourceId),
+        siteId,
+      })
+
+      if (!fullResource) {
+        throw new TRPCError({
+          code: "PRECONDITION_FAILED",
+          message:
+            "Please ensure you are attempting to publish a page that exists",
+        })
+      }
+
+      const version = await incrementVersion({
+        tx,
+        siteId,
+        resourceId,
+        userId,
+      })
+
+      const previousVersion = await tx
+        .selectFrom("Version")
+        .where("Version.versionNum", "=", Number(version.versionId) - 1)
+        .where("Version.resourceId", "=", resourceId)
+        .select("Version.id")
```

Comment:
same: should this be `Version.versionNum` instead?

Reply:
ah, no! we log the `Version.id` below